### PR TITLE
Added IdentityProviders to exports from SignInPage

### PR DESCRIPTION
**Pull request by liam-mackie**
Signed-off-by: Liam Mackie <liam.mackie@octopus.com>

## Hey, I just made a Pull Request!

When using the SignInPage component, the `IdentityProviders` type is passed in, but not exported. This ended up meaning when we were introducing logic to pass through the `guest` provider in our development environments based on environment variables, I needed to introduce an inline type. 

This PR exports the `IdentityProviders` type so that it can be used.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/selfish-actors-walk.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/core-components': patch
+---
+
+Exported IdentityProviders
```
Comment (jhaals) [inline .changeset/selfish-actors-walk.md:5]:
```suggestion
Exported `IdentityProviders` type.
```

**Review by jhaals — COMMENTED** [review]
Thanks! This sounds like something we should export.
The changes to the API need to be added to this PR, you can generate a new API report by running `yarn build:api-reports`

**Comment by benjdlambert** [comment]
Hey :wave: Can we see some example code on how you're planning to use this type? Just not sure why the type needs to be exported and maybe I'm missing something :pray:

**Comment by liam-mackie** [comment]
> Hey 👋 Can we see some example code on how you're planning to use this type? Just not sure why the type needs to be exported and maybe I'm missing something 🙏

Thanks for the review! An example of its use would be while developing without access to the primary authentication method:
```
const enabledProviders: IdentityProviders = [microsoftLoginProvider];
if (process.env.NODE_ENV == 'development') {
  enabledProviders.push('guest');
}

...

SignInPage: props => (
  <SignInPage {...props} auto providers={enabledProviders} />
),
```

This usage would require the use of an inline type:
```
type IdentityProviders = ('guest' | 'custom' | SignInProviderConfig)[];
```

We could alternatively use a ternary in the prop to get the same functionality, but I feel that this may be a useful type to export in future regardless 😄

File: packages/core-components/api-report.md [hunk 2]
```diff
@@ -510,6 +510,11 @@ export type IconLinkVerticalProps = {
   title?: string;
 };
 
+// Warning: (ae-missing-release-tag) "IdentityProviders" is exported by the package, but it is missing a release tag (@alpha, @beta, @public, or @internal)
```
Comment (benjdlambert) [inline packages/core-components/api-report.md:513]: I think the `IdentityProviders` type will need a `/** @public **/` tsdoc notation to remove this warning :)

File: .changeset/selfish-actors-walk.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/core-components': patch
```
Comment (benjdlambert) [inline .changeset/selfish-actors-walk.md:2]: Think this should be a minor change as we've added something to the api surface.

**Review by benjdlambert — COMMENTED** [review]
Thanks! Just two small nits :)

File: .changeset/selfish-actors-walk.md (hunk 3 above)
Comment (benjdlambert) [reply .changeset/selfish-actors-walk.md:2]: You are correct 👍

File: packages/core-components/api-report.md (hunk 2 above)
Comment (liam-mackie) [reply packages/core-components/api-report.md]: Thanks! I've updated that now!

**Review by benjdlambert — APPROVED** [review]
Thanks for this! :tada:

**Comment by suuus** [comment]
Hey @liam-mackie, congrats on your first merged PR 🎉 Nice work 👍 Please let me know if we can be helpful in any way.

**Pull request merged**